src/routes/files.ts

```diff
@@ -1,21 +1,31 @@
-import { Router } from 'express';
+import express, { Router } from 'express';
 import { UploadedFile } from 'express-fileupload';
-import express from 'express';
 
+import { uuidv4 } from '../common';
 import { config } from '../config/config';
 import Message from '../models/message';
 import { sendMessageToApi } from '../service/apiService';
 import { getChat, persistChat, saveFile } from '../service/dataService';
+import { hasSpecialCharacters } from '../service/fileService';
 import { appendSignatureToMessage } from '../service/keyService';
 import { getMyLocation } from '../service/locationService';
 import { parseMessage } from '../service/messageService';
 import { sendEventToConnectedSockets } from '../service/socketService';
 import { getFullIPv6ApiLocation } from '../service/urlService';
 import { FileMessageType, MessageTypes } from '../types';
-import { hasSpecialCharacters } from '../service/fileService';
+import { Path, saveTempFile } from '../utils/files';
 
 const router = Router();
 
+router.post('/upload', async (req: express.Request, res: express.Response) => {
+    const file = req.files.file as UploadedFile;
+    const fileId = uuidv4();
+    const tempPath = new Path(`${fileId}`, '/appdata/tmp');
+
+    await saveTempFile(tempPath, file);
+    res.json(fileId);
+});
```

Comment:
no checks if a file can be saved

Reply:
It checks if the path exists in the 'saveTempFile' function
![image](https://user-images.githubusercontent.com/36887810/170940269-8a02a8cb-d104-4c42-90aa-1d2d225f3894.png)